PropertyChanged.Fody/CollectionCustomAttributeExtension.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Mono.Cecil;
+using Mono.Collections.Generic;
+
+public partial class ModuleWeaver
+{
+    public void MarkAsGeneratedCode(Collection<CustomAttribute> customAttributes)
+    {
+        AddCustomAttributeArgument(customAttributes);
+        AddDebuggerNonUserCodeAttribute(customAttributes);
+    }
+
+    void AddDebuggerNonUserCodeAttribute(Collection<CustomAttribute> customAttributes)
+    {
+        var debuggerAttribute = new CustomAttribute(DebuggerNonUserCodeAttributeConstructor);
+        customAttributes.Add(debuggerAttribute);
+    }
+
+    void AddCustomAttributeArgument(Collection<CustomAttribute> customAttributes)
+    {
+        var version = typeof(ModuleWeaver).Assembly.GetName().Version.ToString();
+        var name = typeof(ModuleWeaver).Assembly.GetName().Name;
```

Comment:
Could you cache these two values in static fields? That `Assembly.GetName()` calls could get expensive if many methods get injected.

Reply:
Done.